### DISCUSSION TITLE
Finetuning a model using weights from pre-trained model that has different input

### DISCUSSION THREAD

**Discussion by Meelisha** [neuralhydrology/neuralhydrology#59] [Q&A] [answered]
I am trying to finetune a cudalstm model without static attributes and initialize the model weights with a pre-trained model that had static attributes for its training. So my question is.. is it possible to use model weights, obtained by training a model eg. ealstm, to finetune a different model that doesn't use static attributes?

**Comment by kratzert** [comment, marked as answer]
Short answer: No!

Generally, finetuning means to adapt the weights of a pre-trained network to a smaller (or slightly different) data set. In some applications, like computer vision, you sometimes only use parts of the pre-trained model (e.g. the first n layer) and train others from scratch, or even change the rest of the architecture. Here  however, especially with the LSTM, you are required to provide the same inputs during finetuning as during pre-training. These features are concatenated and use as a single, combined input vector and the model learned a specific connection between these input features. Although you theoretically could strip away the weights from the input to hidden matrix that corresponds to the static inputs, I highly doubt that this will yield anything usable.

If you do not have the same static attributes for your finetuning task, why not exclude the static attributes from the pretraining as well?

**Reply by Meelisha** [reply]
Right! I understand. I did try the approach and gave me a cell size error. The thing is, I had initially trained an EALSTM model with around 20 epochs taking data for all the 531 basins in the CAMELS, but, it took really long to compute. I wanted to skip this step and train a model with generic dataset which lacked static attributes. Thank you very much for your quick response. I will try to obtain static attributes same as used for pre-training and if that doesn't work I'll pretrain as you have suggested.

Thank you!
